### PR TITLE
fix: skip general response scanning on empty tools/list responses

### PR DESCRIPTION
**Pull request by luckyPipewrench**
## Summary
- Fix false positive where general injection scanner ran on tools/list responses with empty or all-unnamed tool arrays
- Add `isToolsListResult()` that checks for a JSON array in the `tools` field, separate from tool-definition parsing
- Malformed `tools` values (string, object, number, bool) correctly fall through to general scanning — prevents injection bypass

Closes the pen test finding: "MCP tools/list always returns prompt injection detected even for empty requests."

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved detection and handling of tool-related responses, including proper support for empty tool lists.
  * Added validation guards to prevent misclassification of non-tool responses.

* **Tests**
  * Added comprehensive test coverage for tool response recognition and edge case handling.

### CODE REVIEW

File: internal/mcp/tools/tools.go [hunk 1]
```diff
@@ -544,6 +545,28 @@ func collectDescriptions(obj map[string]interface{}, result *[]string, depth int
...
+	if err := json.Unmarshal(result, &probe); err != nil {
+		return false
+	}
+	// json.RawMessage("null") is non-nil in Go — must check string value.
+	// Only treat as tools/list if tools is a JSON array (including empty []).
+	// A string or object in the tools field is malformed and must NOT suppress
+	// general response scanning — otherwise an attacker hides injection there.
+	trimmed := bytes.TrimSpace(probe.Tools)
+	return len(trimmed) > 0 && trimmed[0] == '['
+}
```
[the remark this answers is not in the document]
Comment (luckyPipewrench) [reply internal/mcp/tools/tools.go:584]: Fixed in d98a120. Array elements now verified as JSON objects (first byte '{''). Added tests for string arrays, number arrays, and mixed arrays — all return false.

File: internal/mcp/tools/tools_test.go [hunk 2]
```diff
@@ -75,6 +75,52 @@ func TestTryParseToolsList_Empty(t *testing.T) {
...
+			}
+		})
+	}
+}
+
+func TestScanTools_EmptyToolsList_IsToolsList(t *testing.T) {
+	// An empty tools/list response should set IsToolsList=true so the
+	// general response scanner skips it (avoids false positives).
+	sc := scanner.New(config.Defaults())
+	defer sc.Close()
```
[the remark this answers is not in the document]
Comment (luckyPipewrench) [reply internal/mcp/tools/tools_test.go]: Fixed in d98a120. Switched to testScanner(t).
